blogging/views.py

```diff
@@ -0,0 +1,28 @@
+from django.shortcuts import render
+
+# Create your views here.
+from django.http import HttpResponse, HttpResponseRedirect, Http404
+from django.template import loader
+from blogging.models import Post
+from django.views.generic.list import ListView
+from django.views.generic.detail import DetailView
+
+class BlogListView(ListView):
+    model = Post
+    queryset = Post.objects.order_by('-published_date').exclude(published_date=None)
+    template_name = 'blogging/list.html'
+
+class BlogDetailView(DetailView):
+    model = Post
```

Comment:
I think you need to limit unpublished posts here?
i.e.
`queryset = Post.objects.exclude(published_date__exact=None)`